Moved Contributors' Guide to CONTRIBUTING.rst file

drdenzy opened the pull request:
Fixed #10187

Moved the main content of Contributors' guide to CONTRIBUTING.rst file.

This is to eliminate duplication and unify Airflow's contributions
guidelines in CONTRIBUTING.rst file. The aim is to make CONTRIBUTING.rst
doc a single source of truth for all Airflow Contributors.

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

drdenzy committed:
Moved Contributors' Guide to CONTRIBUTING.rst file

The main content of Contributors' guide to CONTRIBUTING.rst file.

This is to eliminate duplication and unify Airflow's contributions
guidelines in CONTRIBUTING.rst file. The aim is to make CONTRIBUTING.rst
doc a single source of truth for all Airflow Contributors.

drdenzy committed:
Removed trailing whitespaces and fixed typos.

The is a follow up commit to "Moved Contributors' Guide to
CONTRIBUTING.rst file" commit.

RosterIn commented on CONTRIBUTING.rst:
```diff
@@ -20,9 +20,59 @@
+
+  - This is the best way to get a Committer to answer a question.
+
+- Asking a question on `<https://gitter.im/apache/incubator-airflow>`_
```
Gitter? I don't think airflow use this any more

drdenzy replied:
Oh! I will remove that then.

drdenzy committed:
Removed link to defunct Airflow Gitter handle

The is a follow up commit to "Moved Contributors' Guide to
CONTRIBUTING.rst file" commit.

eladkal commented on CONTRIBUTING.rst:
```diff
@@ -20,9 +20,55 @@
+- submitting Pull Requests (PRs) for bug fixes and new features.
+- attending Airflow meets-up events near you:
+
+  - search for Airflow `Meetup <https://www.meetup.com/>`_
```
I wouldn't say that attending to a meetup considered contributing.

potiuk replied:
Agree. Speaking at a meetup - yes. But attending is not enough :).

eladkal commented on CONTRIBUTING.rst:
```diff
@@ -20,9 +20,55 @@
+
+  - search for Airflow `Meetup <https://www.meetup.com/>`_
+
+- helping to answer questions on gitter, the dev email list, PRs, and Jiras.
```
Gitter and Jira are not used any more.

potiuk replied:
Indeed. We have Slack, devlist, Discussions in GitHub, Gitihub Issues and PRs.

eladkal commented on CONTRIBUTING.rst:
```diff
@@ -20,9 +20,55 @@
+
+- Sending an email to: `<dev@airflow.incubator.apache.org>`_
+
+  - This is the best way to get a Committer to answer a question.
```
We do not encourage asking troubleshooting questions in the dev mailing list.
The dev mailing list is to discuss the development of the project, voting on releases/AIPs.
Troubleshooting questions can be asked on troubleshooting channel in Slack or in Stackoverflow

potiuk replied:
Agree with @eladkal. We also have discussions in Github Discussions and those are even better than Slack due to visibility and searchability so we should encourage those even more than #troubleshooting channel.

eladkal commented on CONTRIBUTING.rst:
```diff
@@ -20,9 +20,55 @@
+
+  - This is the best way to get a Committer to answer a question.
+
+- Open a JIRA ticket when you discover a bug and propose a PR with a fix.
```
We moved from Jira to GitHub issues.

eladkal commented on CONTRIBUTING.rst:
```diff
@@ -20,9 +20,55 @@
+
+If so, then add your company name (along with your Github handle) as links to the
+`README.md <https://github.com/apache/airflow/blob/master/README.md#who-uses-apache-airflow>`_
+and submit a PR (with associated JIRA). We will be happy to merge it.
```
The file is [INTHEWILD.md](https://github.com/apache/airflow/blob/master/INTHEWILD.md)
Would be better to refer to https://github.com/apache/airflow/issues/12012

potiuk replied:
Indeed!

potiuk commented on CONTRIBUTING.rst:
```diff
@@ -20,9 +20,55 @@
+
+- Open a JIRA ticket when you discover a bug and propose a PR with a fix.
+
+  - This is the only way to get PRs approved or to report bugs.
```
We cna remove this part.

drdenzy committed:
resolved review conservations
